### ISSUE TITLE
Windows version includes files synced from OneDrive/Sharepoint in space calculations

### ISSUE THREAD

**Issue by theonetruetom** [dundee/gdu#265]
I use this to do disk space analysis on client workstations without disturbing their work (our remote access tool has a PowerShell that runs as SYSTEM, so I can do this without having to remote in physically to each machine), but lately it seems like gdu is including space consumed by files that exist in SharePoint or OneDrive, which throws off my investigation considerably.  

Is there any way to make the app ignore files that don't exist locally and are only cloud synced?

**Comment by btrfs-d** [comment]
I have this same exact issue. @dundee Please advise. This is the best utility I've found for remote storage investigations but the SharePoint/OneDrive issue is incredibly problematic.

**Comment by dundee** [comment]
I am not sure if I we are able to detect whether the file is local or "virtual" and placed in cloud.

Can you use `-i` or `-I` options to ignore the paths?

**Comment by btrfs-d** [comment]
That's problematic because the intent is to see how much local space is being occupied by locally downloaded SharePoint/OneDrive files.

Consider the following scenario (incredibly common with my client environments, and I imagine many others):

 - A workstation contains a 512GB disk.
 - The Windows system and installed applications occupy 75GB of storage on the disk.
 - The user has a OneDrive account logged in and synced, with 40GB of total files in their cloud storage for OneDrive, and 10GB of those downloaded locally.
 - The user also has a company SharePoint library synced locally, being 300GB total, with 50GB of that downloaded locally.
 
 gdu should detect a total of 135GB/512GB storage used (75GB + 10GB + 50GB) on the disk. 
 Currently, this does not work, and gdu will detect as 415GB/512GB (75GB + 40GB + 300GB). Omitting the OneDrive/SharePoint directories defeats the purpose of using a disk scanning utility.

**Comment by theonetruetom** [comment]
The more extreme examples of this are where a 2TB SharePoint is synced with the machine with the same 512GB disk, and it becomes very confusing to do space usage analysis.

**Comment by dundee** [comment]
I will try to check if there are some possibilities to detect this.

**Comment by theonetruetom** [comment]
It looks like there are Windows filesystem attributes, specifically "RecallOnAccess" that, if set, appears to indicate that files are not stored locally.  I don't know how hard it would be to programmatically check for this, but it's a thought.

**Comment by EntropicNinja** [comment]
I'd like to second (3rd) this feature request.

**Comment by ErrorPower2001** [comment]
> Can you use `-i` or `-I` options to ignore the paths?

No, whether it's `-i` or `-I`, `C:/Users/name/OneDrive` or `C:\Users\name\OneDrive` or `C:\\Users\\name\\OneDrive`, with `""` or `''`

**Comment by not-revan** [comment]
I'd also like to second this. This is an amazing utility. I'm trying to automate a workflow where a gdu report is run and appended to low disk space alerts when they're created. The only thing that's stopping me from using gdu for this is knowing it will report on any Onedrive/Sharepoint file, even if they're online only.

**Comment by Florke64** [comment]
> > Can you use `-i` or `-I` options to ignore the paths?
> 
> No, whether it's `-i` or `-I`, `C:/Users/name/OneDrive` or `C:\Users\name\OneDrive` or `C:\\Users\\name\\OneDrive`, with `""` or `''`

Neither can I. @ErrorPower2001 are you getting any error? On Windows, I am getting ```
Error: error parsing regexp: invalid escape sequence: `\U`
``` for `gdu -I 'OneDrive'` which works fine on Linux. This could be sort of workaround, but I had no luck using `-I` switch on Windows.

**Comment by ErrorPower2001** [comment]
> > > Can you use `-i` or `-I` options to ignore the paths?
> > 
> > 
> > No, whether it's `-i` or `-I`, `C:/Users/name/OneDrive` or `C:\Users\name\OneDrive` or `C:\\Users\\name\\OneDrive`, with `""` or `''`
> 
> Neither can I. [@ErrorPower2001](https://github.com/ErrorPower2001) are you getting any error? On Windows, I am getting `` Error: error parsing regexp: invalid escape sequence: `\U` `` for `gdu -I 'OneDrive'` which works fine on Linux. This could be sort of workaround, but I had no luck using `-I` switch on Windows.

yes and no. This issue occurs because C:\Users is being interpreted as C:`\U`sers, which appears to be a problem with gdu's path string parsing on the Windows platform.

**Comment by tpill90** [comment]
Hi All,

I've opened up a PR for this https://github.com/dundee/gdu/pull/609, and tested this on my own setup to confirm that it works as intended.  @dundee I'm not completely familiar with the codebase, but this appeared to do the trick.  Please let me know how/where this should be changed and I'd be more than happy to get this into a good state that you're ready to merge.  Thank you!

**Comment by tpill90** [comment]
I also put up a test build of my PR as a release on my fork, if anyone would like to test it themselves.  You can find it here https://github.com/tpill90/gdu/releases/tag/pr-build

**Comment by tpill90** [comment]
The fix for this was merged and published recently @theonetruetom.  Give it a shot and let me know what you think!

**Comment by ErrorPower2001** [comment]
I completely forgot that I had participated in this issue. In versions prior to gdu, in order to ignore the OneDrive directory, you had to use a string that did **not** include a trailing slash at the end of the directory path, like this:

`pwsh> gdu --ignore-dirs-pattern "C:\\Users\\<userprofile>\\OneDrive" C:\`

And the directory separator could only be double backslashes.

Of course, this only roughly excludes the entire OneDrive directory, but it was sufficient for the time being.
